src/codegen.rs

```diff
@@ -127,14 +164,79 @@ impl<'ink> CodeGen<'ink> {
             annotations,
             &index,
             &mut self.debug,
-            self.got_layout_file.clone(),
         );
 
         //Generate global variables
         let llvm_gv_index =
             variable_generator.generate_global_variables(dependencies, &self.module_location)?;
         index.merge(llvm_gv_index);
 
+        // Build our GOT layout here. We need to find all the names for globals, programs, and
+        // functions and assign them indices in the GOT, taking into account prior indices.
+        let program_globals = global_index
+            .get_program_instances()
+            .into_iter()
+            .fold(Vec::new(), |mut acc, p| {
+                acc.push(p.get_name());
+                acc.push(p.get_qualified_name());
+                acc
+            });
+        let functions = global_index.get_pous().values()
+            .filter_map(|p| match p {
+                PouIndexEntry::Function { name, linkage: LinkageType::Internal, is_generated: false, .. }
+                | PouIndexEntry::Function { name, linkage: LinkageType::Internal, .. } => Some(name.as_ref()),
```

Comment:
maybe I'm not thinking hard enough but isn't that or-pattern redundant? if we are filtering on all `PouIndexEntry::Function`s that have internal linkage and which have `is_generated` set to false, but also on all the `PouIndexEntry::Function`s where that last condition isn't necessarily upheld (and is by thus true), then I feel like the first part of the pattern isn't necessary? 
```suggestion
                PouIndexEntry::Function { name, linkage: LinkageType::Internal, .. } => Some(name.as_ref()),
```

Reply:
Ah good catch the second line should be `PouIndexEntry::FunctionBlock`